docs/instrumentation_examples.md

```diff
@@ -0,0 +1,214 @@
+probe = get_probe("my_attack")
+
+class CustomAttack(CARLADapricotPatch):
+    def _augment_images_with_patch(self, **kwargs):
+        return_value = super()._augment_images_with_patch(**kwargs)
+        x_patch, patch_target, transformations = return_value
+        probe.update(attack_output=x_patch)
+
+        return return_value
+```
+#### Step 1
+This step is the same as before, except `Probe` name is set to`"my_attack"` in line 3, which is what the user can refer to later to apply processing functions through a `Meter` object.
+
+#### Step 2
+The only difference between `CustomAttack` and `CARLADapricotPatch` is that `_augment_images_with_patch` in line 6 has been redefined to call on `CARLADapricotPatch._augment_images_with_patch` and then have `probe` update the value for `x_patch` that results from that call (line 7-9). An updated value of `x_patch` is stored in `attack_output` via `probe.update(attack_output=x_patch)`. Like the `Probe` name `"my_attack"`, `attack_output` can be referenced by the user later to apply additional processing functions through a `Meter` object.
+
+### `Meter` Example Code
+Now that a `Probe` instance has been created, we need to create a `Meter` object to accept any updated values from `Probe` and apply further processing that the user desires. Suppose the user created a script located at `armory/user_init.py` (Please refer to [User Initialization](./scenarios.md#user-initialization) for more details about using the `user_init` block):
+```python
+from armory.instrument import get_hub, Meter
+
+def set_up_meter():
+    meter = Meter(
+        "my_arbitrary_meter_name", lambda x: x, "my_attack.attack_output"
+    )
+    get_hub().connect_meter(meter)
+```
+#### Step 3
+As before, the `Meter` accepts 3 inputs (line 5): a meter name, a metric/function for processing, and a argument name to pass the metric/function.
+- The meter name (`"my_arbitrary_meter_name"`) can be arbitrary within this context
+- Again, `Meter` will record variables monitored by `Probe` as-is (thus `lambda x: x`)
+- The argument passed to the metric/function follows a `.`-separated format (`"my_attack.attack_output"`), which needs to be consistent with `Probe` setup earlier:
+    - `my_attack` matches input in `probe = get_probe("my_attack")`
+    - `attack_output` matches variable name in `probe.update(attack_output=x_patch)`
+
+#### Step 4
+Again, `get_hub().connect_meter(meter)` in line 7 is necessary for establishing the connection between `meter` created in `armory/user_init.py` and `probe` created in `armory/custom_attack.py`.
+
+### Config Setup
+Last but not least, the config file passed to `armory run` needs to be updated for these changes to take effect. Assuming the `"attack"` block has been changed appropriately, the other block that needs to be added is `"user_init"` (please refer to [User Initialization](./scenarios.md#user-initialization) for more details about using the `user_init` block):
```

Comment:
attack field shouldn't need to be "changed". We're already assuming the user is pointing to the attack that the probe is added to

Reply:
resolved with 8cf85780